online-shop/src/components/ShoppingCart.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import '../productModeling/shoppingcart.css';
+import { IProduct } from '../App.js';
+
+
+
+interface IProps {
+    data: IProduct[];
+ 
+}
+interface IState{
+    cartProducts :IProduct[];
+}
+
+
+export class ShoppingCart extends React.Component<IProps,IState>{
+    productsFromCart:any;
+    constructor(props:IProps){
+        super(props);
+        this.state={
+            cartProducts:this.props.data
+        }
+    
+        
+    }
+    render() {
+        if(this.state.cartProducts!=null){
+              this.productsFromCart= this.state.cartProducts.map(
```

Comment:
Simply use a local variable instead of a class attribute for the `productsFromCart`.